crates/services/src/services/share.rs

```diff
@@ -304,6 +310,60 @@ fn build_ws_headers(session: &ClerkSession) -> WsResult<Vec<(HeaderName, HeaderV
     Ok(headers)
 }
 
+fn spawn_auth_token_refresh(client: WsClient, sessions: ClerkSessionStore) {
+    tokio::spawn(async move {
+        let result: WsResult<()> = async {
+            let close_rx = client.subscribe_close();
+            loop {
+                let session_fut = sessions.wait_for_active();
```

Comment:
Will the token refresh hang until there is an active token? Does this mean that the websocket will stay open?